packages/backend-proxy-middleware/src/base/proxy.ts

```diff
@@ -399,13 +399,13 @@ async function updateProxyConfigFromStore(
     // check if system credentials are stored in the store
     try {
         const systemStore = await getService<BackendSystem, BackendSystemKey>({ logger, entityName: 'system' });
-        const system = (await systemStore.read(
-            new BackendSystemKey({ url: localBackend.url, client: localBackend.client })
-        )) ?? {
-            name: '<unknown>',
-            url: localBackend.url,
-            authenticationType: localBackend.authenticationType
-        };
+        const system =
+            (await systemStore.read(new BackendSystemKey({ url: localBackend.url, client: localBackend.client }))) ??
+            ({
+                name: '<unknown>',
+                url: localBackend.url,
+                authenticationType: localBackend.authenticationType
```

Comment:
This needs validation as the yaml file writer does not use the same types. Casting can cause errors downstream of this since it used to be a string but is now a required type. This can be tested by changing the value in ui5.yaml to something other than a supported string value (we have seen this in customer apps) and previewing.